WordPress/gutenberg
# Mobile: Links UI using BottomSheet component

## Description
This PR updates the Links UI to use the BottomSheet component.
In this way, the UI will be updated as requested by @iamthomasbishop here: 
https://github.com/wordpress-mobile/gutenberg-mobile/pull/275#issuecomment-464078628

<img width="933" alt="image" src="https://user-images.githubusercontent.com/1200432/52863801-5eb11200-30fe-11e9-90ea-91e2e12c196c.png">

![links-ui](https://user-images.githubusercontent.com/9772967/53094722-40428080-351b-11e9-81b0-76497b8892a8.png)

**Note:**
- There is a bug on Android when `autoFocus` is active, that will make the BottomSheet hide under the Keyboard. I ~fixed~ hided this issue by not using `autoFocus` on Android. I believe that the user experience will be better that way until we manage to fix this problem.

- Following the design, we removed the header, thus also the "Done" button. Now by dismissing the BottomSheet, it will automatically save any change.
(cc  @iamthomasbishop)

To test:
- Create a paragraph block.
- Check press the `link` button in the format tool bar.
- Check that the UI looks good and the feature works properly.
